src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ReturnCountCheck.java

```diff
@@ -227,10 +234,12 @@ private void visitReturn(DetailAST ast) {
         }
 
         /**
-         * Increase the number of return statements.
+         * Increase the number of return statements and set context return type.
          * @param maxAssigned Maximum allowed number of return statements.
+         * @param voidReturn Identifies if context is void.
          */
-        public void visitLiteralReturn(int maxAssigned) {
+        public void visitLiteralReturn(int maxAssigned, Boolean voidReturn) {
+            context.isVoidContext = voidReturn;
```

Comment:
Why do you set property to the outer class field? Please refer directly to `isVoidContext` from inside the same object:
```java
isVoidContext = voidReturn;
```

Reply:
done